src/polygon/handlers/marketplace.ts

```diff
@@ -216,3 +220,109 @@ export function handleOrderCancelled(
     );
   }
 }
+
+export async function handleTraded(
+  ctx: Context,
+  event: TradedEventArgs,
+  block: BlockData,
+  transaction: Transaction & { input: string },
+  storedData: PolygonStoredData,
+  inMemoryData: PolygonInMemoryState
+): Promise<void> {
+  const addresses = getAddresses(Network.MATIC);
+  const { _trade } = event;
+  const { accounts, nfts } = storedData;
+
+  const assetType = Number(event._trade.sent[0].assetType);
+  const collectionAddress = _trade.sent[0].contractAddress;
+  const marketplaceV3Contract = new MarketplaceV3ABI.Contract(
+    ctx,
+    block.header,
+    addresses.MarketplaceV3
+  );
+  const feesCollector = await marketplaceV3Contract.feeCollector();
+  const feeRate = await marketplaceV3Contract.feeRate();
+  const royaltiesRate = await marketplaceV3Contract.royaltiesRate();
+
+  // NFT
+  if (assetType === 3) {
```

Comment:
do you think it makes sense to store the assetType = 3 || 4 into an enum?

Reply:
+1